tools/dgeni/processors/categorizer.js

```diff
@@ -171,6 +171,29 @@ function getDirectiveOutputAlias(doc) {
   return isDirectiveOutput(doc) ? doc.decorators.find(d => d.name == 'Output').arguments[0] : '';
 }
 
+function getDirectoveSelectors(classDoc) {
+  let blacklist = [
+    '[portal]',
+    '[portalHost]',
+    'textarea[md-autosize]',
+    '[overlay-origin]',
+    '[connected-overlay]',
+  ];
+  let metadata = classDoc.decorators
+    .find(d => d.name === 'Component' || d.name === 'Directive').arguments[0];
+
+  let selectorMatches = /selector\s*:\s*(?:"|')([^']*?)(?:"|')/g.exec(metadata);
+  selectorMatches = selectorMatches && selectorMatches[1];
+
+  if (selectorMatches) {
+    selectorMatches = selectorMatches.split(/\s*,\s*/);
+    selectorMatches = selectorMatches
+      .filter(match => match !== '' && !match.includes('mat') && blacklist.indexOf(match) === -1);
+  }
+
+  return selectorMatches;
```

Comment:
If you immediately return (as with comment above) then you can just remove that.